### ISSUE TITLE
UnnecessaryParentheses not reported for array access, field access and method calls

### ISSUE THREAD

**Issue by zbynek** [checkstyle/checkstyle#17554]
I have read check documentation: https://checkstyle.org/checks/coding/unnecessaryparentheses.html#UnnecessaryParentheses
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac YOUR_FILE.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="tabWidth" value="4"/>
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <module name="TreeWalker">
        <module name="UnnecessaryParentheses">
        </module>
    </module>
</module>


/var/tmp $ cat YOUR_FILE.java
public class Main {

    private static int foo;
    private static Main INSTANCE;
    private String name;

    public static void main(String[] args) {
        foo = (args[0]).length();
        foo = (String.valueOf(foo)).length();
        foo = (INSTANCE.name).length();
        foo = (new String()).length();
        System.out.println(foo);
    }

}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java
Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**

All 4 assignments to `foo` should trigger `UnnecessaryParentheses`

**Comment by romani** [comment]
> checkstyle-X.XX-all.jar

if you used 5.0 version of checkstyle, please be aware that 15 years passed from that version and a lot was improve , please use latest, we accept issues only for most latest released version

**Comment by zbynek** [comment]
@romani sorry, forgot to update that part of the template, this is with 10.26.1

**Comment by romani** [comment]
I hope non of this is specific to type-aware limitation that we have https://checkstyle.org/checks/coding/unnecessaryparentheses.html#Notes


whole logic is at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java , all is covered by test in corresponding file.
PR is welcomed. 

Here is the Markdown table converted from the image:

| Level | Operator | Description | Associativity |
| :---: | :---: | :---: | :---: |
| 16 | `()`<br>`[]`<br>`new`<br>`.`<br>`::` | parentheses<br>array access<br>object creation<br>member access<br>method reference | left-to-right |
| 15 | `++`<br>`--` | unary post-increment<br>unary post-decrement | left-to-right |
| 14 | `+`<br>`-`<br>`!`<br>`~`<br>`++`<br>`--` | unary plus<br>unary minus<br>unary logical NOT<br>unary bitwise NOT<br>unary pre-increment<br>unary pre-decrement | right-to-left |
| 13 | `()` | cast | right-to-left |
| 12 | `*` `/` `%` | multiplicative | left-to-right |
| 11 | `+` `-`<br>`+` | additive<br>string concatenation | left-to-right |
| 10 | `<<` `>>`<br>`>>>` | shift | left-to-right |
| 9 | `<` `<=`<br>`>` `>=`<br>`instanceof` | relational | left-to-right |
| 8 | `==`<br>`!=` | equality | left-to-right |
| 7 | `&` | bitwise AND | left-to-right |
| 6 | `^` | bitwise XOR | left-to-right |
| 5 | `\|` | bitwise OR | left-to-right |
| 4 | `&&` | logical AND | left-to-right |
| 3 | `\|\|` | logical OR | left-to-right |
| 2 | `?:` | ternary | right-to-left |
| 1 | `=` `+=` `-=`<br>`*=` `/=` `%=`<br>`&=` `^=` `\|=`<br>`<<=` `>>=` `>>>=` | assignment | right-to-left |
| 0 | `->`<br>`->` | lambda expression<br>switch expression | right-to-left |

**Comment by mohammadawwad** [comment]
Hello, my team and I are working on a software engineering university project where we contribute to an open-source project. We are interested in working on this issue and would like to reserve it if possible so we can implement a fix.

**Comment by mohammadawwad** [comment]
If it could be assigned to us, that would be great, thanks!

**Comment by zbynek** [comment]
@mohammadawwad there is an open PR for this since August:
https://github.com/checkstyle/checkstyle/pull/17575
I don't think this needs more work on the implementation side, what's missing is just approval from the maintainer(s).

**Comment by romani** [comment]
Fix is done by @zbynek .
It took maintainers  alot of time to accept fix, but all gratitude is for @zbynek , without his energy it would be impossible.
